### PR TITLE
fix: display the selected validator

### PR DESCRIPTION
**Pull request by qperrot**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

Given a user who has an active delegation
When his wants to change the validator
Then clicking on ‘Show less’ does not display the selected validator no the one currently delegating

https://github.com/user-attachments/assets/dbe9841a-24d0-484f-b095-2017c7cdcf41

### ❓ Context

- **JIRA or GitHub link**: [LIVE-14501](https://ledgerhq.atlassian.net/browse/LIVE-14501)
[LIVE-14500](https://ledgerhq.atlassian.net/browse/LIVE-14500)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14501]: https://ledgerhq.atlassian.net/browse/LIVE-14501?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-14500]: https://ledgerhq.atlassian.net/browse/LIVE-14500?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/coin-modules/coin-cardano/src/api/getPools.ts [hunk 1]
```diff
@@ -23,13 +23,25 @@ export async function fetchPoolList(
 export async function fetchPoolDetails(
   currency: CryptoCurrency,
   poolIds: Array<string>,
+  ledgerPoolIds?: Array<string>,
 ): Promise<APIGetPoolsDetail> {
+  const currentPoolIds = poolIds.length == 0 ? ledgerPoolIds : poolIds;
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-cardano/src/api/getPools.ts:28]: What if `ledgerPoolIds` undefined? 🤔 
If `poolIds` length is 0, you might do twice the same request if the response is empty.

**Review by sprohaszka-ledger — COMMENTED** [review]
I am not sure to understand what is the logic you want to implement in the `fetchPoolDetails` 🤔

**Review by sprohaszka-ledger — COMMENTED** [review]
Not sure about the implementation of the `fetchPoolDetails`

File: libs/coin-modules/coin-cardano/src/api/getPools.ts (hunk 1 above)
Comment (sprohaszka-ledger) [reply libs/coin-modules/coin-cardano/src/api/getPools.ts:28]: Maybe this is a version that you want to use?
```js
export async function fetchPoolDetails(
  currency: CryptoCurrency,
  poolIds: Array<string>,
  ledgerPoolIds?: Array<string>,
): Promise<APIGetPoolsDetail> {
  const currentPoolIds = poolIds.length == 0 ? ledgerPoolIds : poolIds;
  const { data } = await network<APIGetPoolsDetail>({
    method: "GET",
    url: isTestnet(currency)
      ? `${CARDANO_TESTNET_API_ENDPOINT}/v1/pool/detail`
      : `${CARDANO_API_ENDPOINT}/v1/pool/detail`,
    params: { poolIds: currentPoolIds },
  });
  if (data.pools.length === 0 && ledgerPoolIds) {
    return fetchPoolDetails(currency, ledgerPoolIds);
  }
  return data;
}
```

Comment (qperrot) [reply libs/coin-modules/coin-cardano/src/api/getPools.ts:28]: Yes actually that's what I wanted to do
Thx

**Review by sprohaszka-ledger — APPROVED** [review]

**Pull request merged**